N/Ncurses/build_tarballs.jl

```diff
@@ -88,5 +88,11 @@ dependencies = [
     HostBuildDependency("Ncurses_jll"),
 ]
 
+init_block = raw"""
+if Sys.isunix()
+    ENV["TERMINFO"] = get(ENV, "TERMINFO", joinpath(Ncurses_jll.artifact_dir, "share", "terminfo"))
```

Comment:
This I also adapted from `Xorg_xkeyboard_config` -- it only sets `TERMINFO` to point at the artifact if it wasn't already set. So the user can still set their own `TERMINFO`. 

However, looking at `man terminfo`, it seems the precedence is `TERMINFO > $HOME/.terminfo > TERMINFO_DIRS` so perhaps I should set `TERMINFO_DIRS` here instead?

Reply:
https://github.com/JuliaPackaging/Yggdrasil/issues/455#issuecomment-581026705

Reply:
I am aware of that issue, and in fact I am using the `withenv` workaround myself in `GAP.jl`. However, it is limited: I have code paths the user triggers which could also in the end need this. It is not practical to repeat the `withenv` in all code. So I think addressing this once and forall here makes sense.

Of course there is a hypothetical chance we'll break something. But that's true for almost any change we make. We'll deal with it if we get reports about it? In the worst case we can revert this change here -- of course we'll then (re-)break other things, but such is life.... ?

Reply:
Another option is that I just set the variable globally in `GAP.jl`, then it'll be limited in which code the change affects -- so breakage still is possible, but minimized *shrug*

Reply:
Yeah, especially for `ccall`s into library wrapping every single call with withenv is annoying.